waiter/integration/waiter/saml_authentication_test.clj

```diff
@@ -0,0 +1,140 @@
+(ns waiter.saml-authentication-test
+  (:require [clojure.data.json :as json]
+            [clojure.java.shell :as shell]
+            [clojure.string :as string]
+            [clojure.test :refer :all]
+            [reaver :refer [parse extract select text attr]]
+            [waiter.util.client-tools :refer :all])
+  (:import (java.net URLEncoder)))
+
+(deftest ^:parallel ^:integration-fast test-default-composite-authenticator
+  (testing-using-waiter-url
+    (let [authenticator-kind (get-in (waiter-settings waiter-url) [:authenticator-config :kind])]
+      (when (= "composite" authenticator-kind)
+        (let [{:keys [service-id body]} (make-request-with-debug-info {} #(make-kitchen-request waiter-url % :path "/request-info"))
+              body-json (json/read-str (str body))]
+          (with-service-cleanup
+            service-id
+            (is (= (retrieve-username) (get-in body-json ["headers" "x-waiter-auth-principal"])))))))))
+
+(deftest ^:parallel ^:integration-fast test-token-authentication-parameter-error
```

Comment:
We should have a similar test for on-the-fly to trigger the authentication parameter validation.

Reply:
Do we have an integration test that fails a request when an invalid authentication is provided?

Reply:
what do you mean, a bad SAML assertion? that's handled by a unit test

Reply:
I meant an `x-waiter-authentication=foo` header/token parameter which is unsupported by the authenticator.

Reply:
that's done in core_test.clj

Reply:
Can you link the test here please?

Reply:
waiter.core-test/test-wrap-auth-bypass request-without-existing-auth-default-named-token-with-authentication-header-2

Reply:
waiter.token-test/test-post-failure-in-handle-token-request post:new-service-description:invalid-authentication